pkg/controller/ingress/ingress_controller_test.go

```diff
@@ -64,8 +64,113 @@ var (
 			},
 		},
 	}
+
+	defaultIngressForClusterLocal = &networkingv1alpha1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        name,
+			Namespace:   namespace,
+			UID:         uid,
+			Labels:      map[string]string{serving.RouteNamespaceLabelKey: namespace, serving.RouteLabelKey: name},
+			Annotations: map[string]string{networking.IngressClassAnnotationKey: network.IstioIngressClassName},
+		},
+		Spec: networkingv1alpha1.IngressSpec{
+			Visibility: networkingv1alpha1.IngressVisibilityExternalIP,
+			Rules: []networkingv1alpha1.IngressRule{{
+				Hosts: []string{"test.default.svc.cluster.local"},
+				HTTP: &networkingv1alpha1.HTTPIngressRuleValue{
+					Paths: []networkingv1alpha1.HTTPIngressPath{{
+						Timeout: &metav1.Duration{Duration: 5 * time.Second},
+					}},
+				},
+			}},
+		},
+		Status: networkingv1alpha1.IngressStatus{
+			LoadBalancer: &networkingv1alpha1.LoadBalancerStatus{
+				Ingress: []networkingv1alpha1.LoadBalancerIngressStatus{{
+					DomainInternal: "cluster-local-gateway." + serviceMeshNamespace + ".svc.cluster.local",
+				}},
+			},
+		},
+	}
 )
 
+func TestClusterLocalSvc(t *testing.T) {
+	logf.SetLogger(logf.ZapLogger(true))
+	test := struct {
```

Comment:
Why the entire complicated setup with inner tests and stuff? I think this can be condensed down quite a bit. You don't even need to look at routes in this test.